radix-engine-interface/src/model/resource/non_fungible_local_id.rs

```diff
@@ -73,177 +67,74 @@ impl From<u128> for NonFungibleLocalId {
     }
 }
 
-/// Represents a key for a non-fungible resource
+/// Represents the local id of a non-fungible.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum NonFungibleLocalId {
+    String(String),
     Number(u64),
-    UUID(u128),
     Bytes(Vec<u8>),
-    String(String),
+    UUID(u128),
```

Comment:
I wonder if we should make a UUID actually be a `[u8; 16]`?

Also, in the slack thread, we discussed enforcing that it was _actually_ validated as a UUID, ie that it conforms to the UUID spec for randomly generated UUIDs. I think one example of such an id is this:

```
{00000000-0000-4000-8000-000000000000}
```

NB - I think based on my reading that the UUID is the above, with  the 4 nibble fixed, and the top 2 bits of the 8 are fixed (the bottom 2 are random - ie that hex nibble can be any from 8..b) and the other 0s are all 4-bits of randomness. IE we have (128-4-2 = 122 bits of randomness).

What do you think?

Reply:
Now stick to UUID v4, variant 1.  Don't think it's necessary to use `[u8; 16]` which is the network bytes abstraction of UUID.

Reply:
👍 . Fair enough. It's just that u128 to me suggests it's a number, which it most definitely isn't. But that's really semantics I guess!